packages/rest/src/parser.ts

```diff
@@ -101,22 +102,31 @@ function buildOperationArguments(
       throw new Error('$ref parameters are not supported yet.');
     }
     const spec = paramSpec as ParameterObject;
-    switch (spec.in) {
-      case 'query':
-        paramArgs.push(request.query[spec.name]);
-        break;
-      case 'path':
-        paramArgs.push(pathParams[spec.name]);
-        break;
-      case 'header':
-        paramArgs.push(request.headers[spec.name.toLowerCase()]);
-        break;
-      // TODO(jannyhou) to support `cookie`,
-      // see issue https://github.com/strongloop/loopback-next/issues/997
-      default:
-        throw new HttpErrors.NotImplemented(
-          'Parameters with "in: ' + spec.in + '" are not supported yet.',
-        );
+    const rawValue = getParamFromRequest();
+    const coercedValue = coerceParameter(rawValue, spec.schema);
+    paramArgs.push(coercedValue);
+
+    function getParamFromRequest() {
```

Comment:
Please move this function outside of `buildOperationArguments` and pass any data via arguments, e.g.

```ts
function getParamFromRequest(spec.name, spec.in, request);
```

Reply:
@bajtos any particular reason to move it out other than a cleaner code orginazation? Just curious...since it increases the effort of defining more parameters as the input of `getParamFromRequest` :-p

Reply:
I am not entirely sure about the recent V8 versions, but in the days of Node.js 0.10, V8 was often not able to optimize functions defined inside other functions and relying on closure variables.

Design-wise, by moving `getParamFromRequest` to a top level, it becomes possible to test this function in isolation in unit tests.